### PR TITLE
[refactor] Read Env in running time

### PR DESCRIPTION
**Pull request by Aloento**
## Summary of the Pull Request  

This PR focuses on improving the functionality, performance, and maintainability of the application with the following changes:  

### Changes  

Update entrypoint script to change directory and simplify start command
Update Dockerfile and add entrypoint script; remove nginx configuration

### CODE REVIEW

**Comment by vladimirvshivkov** [comment]
no, it is required app to read environment vars from `process.env.ENV`

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+SD_BACKEND_URL=$SD_BACKEND_URL
+SD_CLIENT_ID=$SD_CLIENT_ID
+SD_AUTHORITY_URL=$SD_AUTHORITY_URL
+SD_REDIRECT_URL=$SD_REDIRECT_URL
+SD_LOGOUT_REDIRECT_URL=$SD_LOGOUT_REDIRECT_URL
+SD_AUTH_SECRET=$SD_AUTH_SECRET
+EOF
+
+pnpm run build
+pnpm start --port 80
```
Comment (sgmv) [inline entrypoint.sh:15]: You shouldn't run your application by `pnpm` in production envitonment.

**Review by sgmv — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -4,20 +4,11 @@ ENV PNPM_HOME="/pnpm"
...
-FROM nginx:stable-alpine
-
-COPY nginx.conf /etc/nginx/conf.d/default.conf
-
-COPY --from=prod /app/dist/ /app
 
 EXPOSE 80
 
-CMD [ "nginx" ]
+CMD [ "cd /app && pnpm run build && pnpm start --port 80" ]
```
Comment (SebastianGode) [inline Dockerfile:14]: Don't use the development server from `farm preview`  in a Dockerfile

**Comment by Aloento** [comment]
> keep original dockerfile

can't, the env will not take effect when using nginx.

File: Dockerfile [hunk 3]
```diff
@@ -4,20 +4,12 @@ ENV PNPM_HOME="/pnpm"
...
-
-COPY nginx.conf /etc/nginx/conf.d/default.conf
-
-COPY --from=prod /app/dist/ /app
+RUN pnpm install -D express
 
 EXPOSE 80
 
-CMD [ "nginx" ]
+CMD [ "cd /app && pnpm run build && node server.js" ]
```
Comment (SebastianGode) [inline Dockerfile:15]: Ok, I'm fine with Express as server now

**Review by SebastianGode — DISMISSED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -4,20 +4,14 @@ ENV PNPM_HOME="/pnpm"
...
+RUN apk add --no-cache nginx
 
 COPY nginx.conf /etc/nginx/conf.d/default.conf
 
-COPY --from=prod /app/dist/ /app
-
 EXPOSE 80
 
-CMD [ "nginx" ]
+CMD [ "cd /app && pnpm run build && nginx -g 'daemon off;'" ]
```
Comment (sgmv) [inline Dockerfile:17]: In that case, you will wait the container startup till your static files will be compiled. I'm not sure that we should do it.

**Review by sgmv — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (Aloento) [reply Dockerfile:17]: env can only be obtained when the container starts, so I have to.

**Review by SebastianGode — APPROVED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Review by sgmv — APPROVED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Pull request merged**